linker-interface/shared/src/main/scala/org/scalajs/linker/interface/LegacyV1_2Adapter.scala

```diff
@@ -0,0 +1,213 @@
+   * because:
+   *
+   * - Source maps do not contain nested objects.
+   * - The file URI should not contain '"', because URI.toASCIIString (which is
+   *   used by the StandardLinker) never returns them.
+   *
+   * So as a legacy mechanism, this is OK-ish. It keeps us from having to build
+   * the infrastructure to parse JSON cross platform.
+   */
+  private val fileFieldRe = """(?m)([,{])\s*"file"\s*:\s*".*"\s*([,}])""".r
+
+  /** Retrieve the linker JS file and an optional source map */
+  private def retrieveOutputFiles(report: Report,
+      outputDirectory: OutputDirectory)(
+      implicit ec: ExecutionContext): Future[(ByteBuffer, Option[ByteBuffer])] = {
+    val outDirImpl = OutputDirectoryImpl.fromOutputDirectory(outputDirectory)
+
+    val module = {
+      if (report.publicModules.size != 1) {
+        throw new UnsupportedLinkerOutputException(
+            "Linking did not return exactly one public module. Full report:\n" +
+            report)
+      }
+
+      report.publicModules.head
+    }
+
+    val checkFiles = for {
+      foundFilesList <- outDirImpl.listFiles()
+    } yield {
+      val foundFiles = foundFilesList.toSet
+      val expectedFiles = Set(module.jsFileName) ++ module.sourceMapName
+
+      if (foundFiles != expectedFiles) {
+        if (expectedFiles.subsetOf(foundFiles)) {
+          throw new UnsupportedLinkerOutputException(
+              "Linking produced more than a single JS file (and source map). " +
+              s"Expected files:\n$expectedFiles\nProduced files:\n$foundFiles")
+        } else {
+          throw new AssertionError(
+              "Linking did not produce the files mentioned in the report. " +
+              "This is a bug in the linker. " +
+              s"Expected files:\n$expectedFiles\nProduced files:\n$foundFiles")
+        }
+      }
+    }
+
+    for {
+      _ <- checkFiles
+      jsFileContent <- outDirImpl.readFull(module.jsFileName)
+      sourceMapContent <- Future.traverse(module.sourceMapName.toList)(outDirImpl.readFull(_))
+    } yield {
+      (jsFileContent, sourceMapContent.headOption)
+    }
+  }
+
+  /** Patches the JS file content to have the provided source map link (or none)
+   *
+   *  Looks for a line of the form `//# sourceMappingURL=.*` and replaces the
+   *  URL with the provided `sourceMapURI`. In case `sourceMapURI` is None, the
+   *  line is replaced with an empty line.
+   */
+  private def patchJSFileContent(content: String,
+      sourceMapURI: Option[URI]): String = {
+
+    val newLine =
+      sourceMapURI.map(u => s"//# sourceMappingURL=${u.toASCIIString}")
+
+    sourceMapRe.findFirstMatchIn(content).fold {
+      content + newLine.fold("")("\n" + _ + "\n")
+    } { reMatch =>
+      val res = new StringBuilder
+
+      res.append(reMatch.before)
+
+      /* If there is no source map link, keep an empty line to not break a
+       * potential (unlinked) source map
+       */
+      newLine.foreach(res.append(_))
```

Comment:
It seems that the code does not in fact keep an empty line when `newLine.isEmpty`.